src/custom/components/CowSubsidyModal/utils.ts

```diff
@@ -1,18 +1,26 @@
 import { BigNumber } from 'bignumber.js'
 import { COW_SUBSIDY_DATA } from './constants'
 
-export function getDiscountFromBalance(balanceAtomsBn: BigNumber) {
-  const [, firstDiscount] = COW_SUBSIDY_DATA[0]
+const LAST_INDEX = COW_SUBSIDY_DATA.length - 1
+const [, firstDiscount] = COW_SUBSIDY_DATA[0]
 
+export function getDiscountFromBalance(balanceAtomsBn: BigNumber) {
   let discount = firstDiscount
   let tier = 0
-  for (const [threshold, thresholdDiscount] of COW_SUBSIDY_DATA) {
-    // is balance < current threshold?
-    // e.g 2443 < TIER 3: >1000? FALSE
-    if (balanceAtomsBn.lt(new BigNumber(threshold))) break
+  // Here we use a sliced verison of our data without index 0 (0 amt tier)
+  // because loop-wise a balance less than or equal to 0 and 100 (indices 0 and 1, respectively) are the same
+  for (const [threshold, thresholdDiscount] of COW_SUBSIDY_DATA.slice(1)) {
```

Comment:
should `COW_SUBSIDY_DATA.slice(1)` be already a constant? (so u don't need to reconstruct the array each time u ask for the tier. It doesn't change

Reply:
done